internal/bundle/plain.go

```diff
@@ -0,0 +1,66 @@
+package bundle
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"k8s.io/cli-runtime/pkg/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PlainV0 struct {
+	Objects []client.Object
+}
+
+func LoadPlainV0(bundle fs.FS) (*PlainV0, error) {
+	const manifestsDir = "manifests"
+	var objects []client.Object
+	dirEntries, err := fs.ReadDir(bundle, manifestsDir)
+	if err != nil {
+		return nil, err
+	}
+	if len(dirEntries) == 0 {
+		return nil, errors.New("invalid bundle: found zero files")
+	}
+	for _, manifest := range dirEntries {
+		if manifest.IsDir() {
+			return nil, fmt.Errorf("subdirectories are not allowed within the %q directory "+
+				"of the bundle image filesystem: found %q", manifestsDir, filepath.Join(manifestsDir, manifest.Name()))
+		}
+		theseObjects, err := getObjects(bundle, manifest)
+		if err != nil {
+			return nil, err
+		}
+		objects = append(objects, *theseObjects...)
+	}
+	if len(objects) == 0 {
+		return nil, errors.New("invalid bundle: found zero objects: " +
+			"plain+v0 bundles are required to contain at least one object")
+	}
+	return &PlainV0{Objects: objects}, nil
+}
+
+func getObjects(bundle fs.FS, manifest fs.DirEntry) (*[]client.Object, error) {
```

Comment:
No need to return a pointer to a slice. A slice is already a pointer (so you can continue returning `nil, err`).
```suggestion
func getObjects(bundle fs.FS, manifest fs.DirEntry) ([]client.Object, error) {
```